app/memory/custom_summary_memory.py

```diff
@@ -0,0 +1,26 @@
+from langchain_core.messages import BaseMessage, SystemMessage
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain.memory import ConversationSummaryMemory
+
+class SummaryChatMessageHistory(BaseChatMessageHistory):
+    def __init__(self, summary_memory: ConversationSummaryMemory):
+        self.summary_memory = summary_memory
+
+    @property
+    def messages(self) -> list[BaseMessage]:
+        if self.summary_memory.buffer:
+            return [SystemMessage(content=self.summary_memory.buffer)]
+        return []
+
+    def add_message(self, message: BaseMessage) -> None:
+        self.summary_memory.chat_memory.add_message(message)
```

Comment:
Add error handling to verify that message is the correct type (BaseMessage or subclass). Unexpected message formats could cause errors downstream.